Tests/test_phenotype.py

```diff
@@ -311,6 +312,26 @@ def test_WellRecord(self):
 
         w[1] = 1
 
+    def test_JsonIterator(self):
+        """Test basic functionalities of JsonIterator file parser."""
+        # Parse file content big enouth to trigger issue #3783
+        JSON = '{"csv_data": {"Plate Type": "PM-999"}, "measurements": {"Hour": 9}}'
+        f = io.StringIO(JSON)
+
+        gen = phenotype.phen_micro.JsonIterator(f)
+        for w in gen:
+            self.assertEqual(w.id, "PM999")
```

Comment:
I wouldn't bother defining the ``JSON`` and ``gen`` variables, this is just as clear to me - assuming I got the black formatting right:

```python
        handle = io.StringIO(
            '{"csv_data": {"Plate Type": "PM-999"}, "measurements": {"Hour": 9}}'
        )
        for w in phenotype.phen_micro.JsonIterator(handle):
            self.assertEqual(w.id, "PM999")
```

What do you think? You could even use a context manager (with statement)?

I'm not keen on ``w`` as a variable name for well record, but this is consistent with the rest of the file so that's OK.

Reply:
I see no problem with adjusting the code according to you suggestions. 

If you want, suggestion for better name for w? I just used the existing code as template.

Reply:
I'm OK with ``w`` given the existing usage in the file.